File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_profit = 0; 
        int min_val = pow(10,4);
        
        
        
        for(int i = 0; i < prices.size(); i++)
        {
            min_val = min(prices[i], min_val);
            max_profit = max(prices[i]-min_val, max_profit);
        }
        
        
        return max_profit; 
        
    }
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        minb, maxs = float("inf"), -1
        max_profit = 0
        
        for x in prices:
            minb = min(minb, x)
            max_profit = max((x-minb), max_profit)
            
        return max_profit
        